docs/linux/external_fuzzing_usbip.md

```diff
@@ -0,0 +1,34 @@
+# **USB/IP Fuzzing for Linux Kernel**
+
+Syzkaller supports fuzzing the Linux kernel USB/IP subsystem externally. We can set up a virtual network and send USB/IP packets to the client kernel as they are being received from an external server.  
+
+USB/IP fuzzing needs USB/IP configurations to be enabled. You can find the list in the configurations part. 
+
+USB/IP Fuzzing consists of 3 main parts:
+1. USB/IP pseudo-syscalls
+2. Syzkaller descriptions.
+3. KCOV changes that allow to collect coverage from background threads and interrupts
+
+### **Configurations:**
+
+Following configurations should be enabled and set for USB/IP. 
+`CONFIG_USBIP_VHCI_HC_PORTS` can be set from 1:15 with default value 8.
+`CONFIG_USBIP_VHCI_NR_HCS` can be set from 1:128 with default value 1.
+
+```
+CONFIG_USBIP_CORE=y
+CONFIG_USBIP_VHCI_HCD=y
+CONFIG_USBIP_VHCI_HC_PORTS=8
+CONFIG_USBIP_VHCI_NR_HCS=16
+CONFIG_USBIP_HOST=y
+CONFIG_USBIP_VUDC=y
+CONFIG_USBIP_DEBUG =y
+```
+
+### **Pseudo-syscalls:**
+
+Currently syzkaller defines 2 USB/IP pseudo-syscalls (See [this](https://github.com/harputluoglu/syzkaller/blob/usbip/executor/common_linux.h#L1560) for psedo-syscalls and [this](https://github.com/harputluoglu/syzkaller/blob/usbip/sys/linux/usbip.txt) for syzkaller descriptions of these pseudo-syscalls):
+
+`Syz_usbip_server_init` : This is for setting up USB/IP server. It creates a TCP connection with the client . It opens a `/sys/devices/platform/vhci_hcd.#proc_id/attach` file which is specific to each USB/IP process. Later, the server needs to write the USB/IP client’s socket descriptor as well as other USB device information into this file so that the USB/IP communication between client and server can start and client’s kernel can receive USB/IP packets from the server. 
```

Comment:
`Syz_usbip_server_init` => `syz_usbip_server_init` here and below.

`client .` => `client.`